contracts/lib/IdentityBase.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.16;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "../interfaces/IState.sol";
+import "../lib/ClaimBuilder.sol";
+import "../lib/OnChainIdentity.sol";
+
+// /**
+//  * @dev Contract managing onchain identity
+//  */
+contract IdentityBase is OwnableUpgradeable {
+    using OnChainIdentity for OnChainIdentity.Trees;
+    using OnChainIdentity for OnChainIdentity.IdentityData;
+
+    OnChainIdentity.IdentityData public identity;
+    OnChainIdentity.Trees internal treeRoots;
+    OnChainIdentity.LastTreeRoots public lastTreeRoots;
+
+    // This empty reserved space is put in place to allow future versions
+    // of the SMT library to add new Data struct fields without shifting down
+    // storage of upgradable contracts that use this struct as a state variable
+    // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)
+    uint256[47] private __gap;
+
+    function getMaxSmtDepth() public pure virtual returns (uint256) {
+        return 40;
+    }
+
+    function initialize(address _stateContractAddr) public virtual initializer {
+        identity.initialize(_stateContractAddr, 
+            getMaxSmtDepth(),
+            treeRoots);
+
+        __Ownable_init();
+    }
+
+    /**
+     * @dev Add claim
+     * @param claim - claim data
+     */
+    function addClaim(uint256[8] calldata claim) public virtual onlyOwner {
+        treeRoots.addClaim(claim);
```

Comment:
It should be `identity.addClaim/revokeClaim/transitState/calcIdentityState/etc` not `treeRoots.addClaim`.